### PR TITLE
Fix SQL error when retry of multipart message

### LINKED ISSUE #311: SQL error when retry of multipart message [$250]
Hi, I see in my log this entry, does anybody know, what does it mean?

```
Thu 2017/05/18 14:38:47 gammu-smsd[544]: Moved to errorbox, reached MaxRetries: 2919
Thu 2017/05/18 14:38:47 gammu-smsd[544]: Error code: 1062, Error: Duplicate entry '2919-1' for key 'PRIMARY'
Thu 2017/05/18 14:38:47 gammu-smsd[544]: SQL failure: 79
Thu 2017/05/18 14:38:47 gammu-smsd[544]: Error writing to database (SMSDSQL_AddSentSMSInfo)
Thu 2017/05/18 14:38:47 gammu-smsd[544]: Error code: 1062, Error: Duplicate entry '2919-2' for key 'PRIMARY'
Thu 2017/05/18 14:38:47 gammu-smsd[544]: SQL failure: 79
Thu 2017/05/18 14:38:47 gammu-smsd[544]: Error writing to database (SMSDSQL_AddSentSMSInfo)
```

<bountysource-plugin>

---
There is a **[$250 open bounty](https://www.bountysource.com/issues/45363539-sql-error-when-retry-of-multipart-message?utm_campaign=plugin&utm_content=tracker%2F253405&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F253405&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gnidorah**
Closes #311
Thank you!

### CODE REVIEW

File: include/gammu-message.h [hunk 1]
```diff
@@ -450,6 +450,10 @@ typedef struct {
 	 */
 	gboolean RejectDuplicates;
 	/**
+	 * Whether to skip message.
+	 */
+	gboolean SkipMessage;
+	/**
```
Comment (nijel) [inline include/gammu-message.h]: I'm not really sure it is good idea to include this in Gammu API as it won't have any use there. This is purely SMSD feature.

**Comment by gnidorah** [comment]
@nijel Done, thanks!

File: smsd/core.h [hunk 2]
```diff
@@ -110,6 +110,7 @@ struct _GSM_SMSDConfig {
 	GSM_SMSC	SMSC, SMSCCache;
 	const char	*skipsmscnumber;
 	int		IgnoredMessages;
+	gboolean 	SkipMessage[GSM_MAX_MULTI_SMS];
```
Comment (nijel) [inline smsd/core.h]: This should be initialized somewhere to avoid possible problems with backends not setting the flag (eg. files).

Comment (gnidorah) [inline smsd/core.h]: @nijel Indeed, sorry.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**